examples/platforms/posix/logging.c

```diff
@@ -73,8 +81,45 @@ OT_TOOL_WEAK void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const
 
     otEXPECT_ACTION(charsWritten >= 0, logString[offset] = 0);
 
+    switch (aLogLevel)
+    {
+    case OT_LOG_LEVEL_NONE:
+        logLevel = LOG_ALERT;
+        logColor = LOG_COLOR_CODE_DEFAULT;
+        break;
+    case OT_LOG_LEVEL_CRIT:
+        logLevel = LOG_CRIT;
+        logColor = LOG_COLOR_CODE_RED;
+        break;
+    case OT_LOG_LEVEL_WARN:
+        logLevel = LOG_WARNING;
+        logColor = LOG_COLOR_CODE_YELLOW;
+        break;
+    case OT_LOG_LEVEL_NOTE:
+        logColor = LOG_COLOR_CODE_CYAN;
+        logLevel = LOG_NOTICE;
+        break;
+    case OT_LOG_LEVEL_INFO:
+        logColor = LOG_COLOR_CODE_GREEN;
+        logLevel = LOG_INFO;
+        break;
+    case OT_LOG_LEVEL_DEBG:
+        logColor = LOG_COLOR_CODE_DEFAULT;
+        logLevel = LOG_DEBUG;
+        break;
+    default:
+        assert(false);
+        logLevel = LOG_DEBUG;
+        break;
+    }
+
 exit:
-    syslog(LOG_CRIT, "%s", logString);
```

Comment:
The reason why everything was mapped to syslog's critical log level is because rsyslog defaults to notice log level. As a result, critical level ensures that all OpenThread logs can be seen without any change to `/etc/rsyslog.d`. Note that `examples/platforms/posix` is purely intended for simulating OpenThread and not to be used for anything beyond simulation. `syslog` was only introduced here as a convenient method for serializing logs between multiple posix simulation processes.